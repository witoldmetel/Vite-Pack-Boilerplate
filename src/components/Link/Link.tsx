import { ReactNode, useState } from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

type LinkProps = {
  page: string;
  children: ReactNode;
};

export const Link = ({ page, children }: LinkProps) => {
  const [status, setStatus] = useState(STATUS.NORMAL);

  const onMouseEnter = () => setStatus(STATUS.HOVERED);
  const onMouseLeave = () => setStatus(STATUS.NORMAL);

  return (
    <a
      className={status}
      href={page || '#'}
      target="_blank"
      rel="noreferrer"
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
};
