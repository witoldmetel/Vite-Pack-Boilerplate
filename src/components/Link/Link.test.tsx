import { Link } from './Link';
import renderer from 'react-test-renderer';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test('Link changes the class when hovered', () => {
  const component = renderer.create(<Link page="http://antfu.me">Anthony Fu</Link>);

  let tree = toJson(component);

  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();

  // re-rendering
  tree = toJson(component);
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseLeave();
  // re-rendering
  tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
