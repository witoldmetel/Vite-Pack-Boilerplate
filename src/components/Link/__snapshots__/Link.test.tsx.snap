// Vitest Snapshot v1

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://www.linkedin.com/in/witoldmetel/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Witold Metel
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://www.linkedin.com/in/witoldmetel/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Witold Metel
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://www.linkedin.com/in/witoldmetel/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  rel="noreferrer"
  target="_blank"
>
  Witold Metel
</a>
`;
